contracts/extensions/StakedExpenditure.sol

```diff
@@ -0,0 +1,299 @@
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.7.3;
+pragma experimental ABIEncoderV2;
+
+import "./../colony/ColonyDataTypes.sol";
+import "./../colonyNetwork/IColonyNetwork.sol";
+import "./../patriciaTree/PatriciaTreeProofs.sol";
+import "./ColonyExtensionMeta.sol";
+
+// ignore-file-swc-108
+
+
+contract StakedExpenditure is ColonyExtensionMeta, PatriciaTreeProofs {
+
+  // Events
+
+  event ExpenditureMadeViaStake(address indexed creator, uint256 expenditureId, uint256 stake);
+  event ExpenditureCancelled(uint256 expenditureId);
+  event StakeReclaimed(uint256 expenditureId);
+
+  // Datatypes
+
+  struct Stake {
+    address creator;
+    uint256 amount;
+  }
+
+  // Storage
+
+  uint256 stakeFraction;
+
+  mapping (uint256 => Stake) stakes;
+
+  // Modifiers
+
+  modifier onlyRoot() {
+    require(colony.hasUserRole(msgSender(), 1, ColonyDataTypes.ColonyRole.Root), "staked-expenditure-caller-not-root");
+    _;
+  }
+
+  // Overrides
+
+  /// @notice Returns the identifier of the extension
+  function identifier() public override pure returns (bytes32) {
+    return keccak256("StakedExpenditure");
+  }
+
+  /// @notice Returns the version of the extension
+  function version() public override pure returns (uint256) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    require(address(colony) == address(0x0), "extension-already-installed");
+
+    colony = IColony(_colony);
+  }
+
+  /// @notice Called when upgrading the extension
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension
+  function deprecate(bool _deprecated) public override auth {
+    deprecated = _deprecated;
+  }
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
+    selfdestruct(address(uint160(address(colony))));
+  }
+
+  // Public
+
+  function setStakeFraction(uint256 _stakeFraction) public onlyRoot {
+    require(_stakeFraction <= WAD, "staked-expenditure-value-too-large");
+    stakeFraction = _stakeFraction;
+  }
+
+  function makeExpenditureWithStake(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _domainId,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    uint256 domainRep = getReputationFromProof(_domainId, _key, _value, _branchMask, _siblings);
+    uint256 stakeAmount = wmul(domainRep, stakeFraction);
```

Comment:
What's the benefit? I guess the scenario is: a domain has zero rep, so there's no limitation on people creating expenditures. But a single wei doesn't seem like it would make much difference there?

Reply:
With the current implementation, if you have zero stake (including when you were not required to stake, either because this product is 0 or stakeFraction is), you cannot cancel the expenditure.

Reply:
Is that the case even after this change:

https://github.com/JoinColony/colonyNetwork/pull/1045/files#diff-8074fb992b15ec398f7df15b088ef3d4af2fb263c3f45541de8e6b452795a468R126

Reply:
Nope, that solves it :+1: